app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -658,12 +667,66 @@ private void initFilterChips() {
                 checkBoxTriStates.setState(CheckBoxTriStates.UNKNOWN);
                 NearbyFilterState.setWlmSelected(isChecked);
                 presenter.filterByMarkerType(nearbyFilterSearchRecyclerViewAdapter.selectedLabels, checkBoxTriStates.getState(), true, false);
+                updatePlaceList(chipNeedsPhoto.isChecked(),
+                    chipExists.isChecked(), chipWlm.isChecked());
             }else{
                 chipWlm.setChecked(!isChecked);
             }
         });
     }
 
+    /**
+     * Updates Nearby place list according to available chip states
+     *
+     * @param NeedsPhoto is chipNeedsPhoto checked
+     * @param isExists is chipExists checked
+     * @param isWlm is chipWlm checked
+     */
+    private void updatePlaceList(final boolean NeedsPhoto, final boolean isExists,
```

Comment:
@neslihanturan I thought the name isNeedsPhoto sounds a little odd that's why I named it like this so that we can discuss a suitable name. Please let me know a suitable name, I will change it.

Reply:
Yeah, isNeedsPhoto is not a nice name either. How about using "needsPhoto" instead of "NeedsPhoto"? Also, with the same logic we can rename isExists to "exists".